### PR TITLE
improve book per audit report

### PR DESCRIPTION
**Pull request by conradoplg**
Closes https://github.com/ZcashFoundation/frost/issues/466

I split the tutorial into multiple subsections to make the structure less confusing. This will make the edits harder to see - sorry. Try using `git diff --color-moved origin/main`.

### CODE REVIEW

File: book/src/tutorial/importing.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+you can use [serde](https://serde.rs/) to serialize any structure that needs
+to be transmitted. Import example:
+
+```
+[dependencies]
+frost-ristretto255 = { version = "0.6.0", features = ["serde"] }
+```
+
+Note that serde usage is optional. Applications can use different encodings, and
+to suppor that, all structures that need to be transmitted have public getters
```
Comment (dconnolly) [inline book/src/tutorial/importing.md:33]:
```suggestion
to support that, all structures that need to be transmitted have public getters
```

File: book/src/tutorial/signing.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+[`round1::commit()`](https://docs.rs/frost-ristretto255/latest/frost_ristretto255/round1/fn.commit.html):
+
+```rust,no_run,noplayground
+{{#include ../../../frost-ristretto255/README.md:round1_commit}}
+```
+
+The `SigningNonces` must be kept by the participant to use in Round 2, while the
+`SigningCommitments` must be sent to the Coordinator using an [authenticated
+channel](https://frost.zfnd.org/terminology.html#broadcast-channel).
+
```
Comment (dconnolly) [inline book/src/tutorial/signing.md:30]: When sending the `round1::SigningCommitments` to the Coordinator, this 'just' needs to be an authenticated peer-to-peer channel, not a broadcast channel, right? 

```suggestion
The `SigningNonces` must be kept by the participant to use in Round 2, while the
`SigningCommitments` must be sent to the Coordinator using an [authenticated
channel](https://frost.zfnd.org/terminology.html#peer-to-peer-channel).

```

Comment (conradoplg) [reply book/src/tutorial/signing.md:30]: Right, wrong link, thanks!

File: book/src/tutorial/signing.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+with the same `SigningPackage` sent to the participants and the
+`PublicKeyPackage` from the key generation (which is used to validate each
+`SignatureShare`).
+
+```rust,no_run,noplayground
+{{#include ../../../frost-ristretto255/README.md:aggregate}}
+```
+
+The returned signature, a `Signature`, will be a valid signature for the message
+chosen in Round 2 for the group verifying key in the `PublicKeyPackage`.
```
Comment (dconnolly) [inline book/src/tutorial/signing.md:91]:
```suggestion
The returned signature, a `Signature`, will be a valid signature for the message
in the `SigningPackage` in Round 2 for the group verifying key in the `PublicKeyPackage`.
```

File: book/src/tutorial/trusted-dealer.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
+# Trusted Dealer Key Generation
+
+The diagram below shows the trusted dealer key generation process. Dashed lines
+represent data being sent through an [authenticated and confidential communication
+channel](https://frost.zfnd.org/terminology.html#peer-to-peer-channel).
+
+![Diagram of Trusted Dealer Key Generation, illustrating what is explained in the text](tkg.png)
```
Comment (dconnolly) [inline book/src/tutorial/trusted-dealer.md:7]: <3 
image

File: book/src/tutorial/trusted-dealer.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+Refer to the [Terminology page](https://frost.zfnd.org/terminology.html#peer-to-peer-channel)
+for more details.
+
+Failure of using a **confidential** channel may lead to the shares being
+stolen and possibly allowing signature forgeries if a threshold number of
+them are stolen.
+
+Failure of using an **authenticated** channel may lead to shares being
+sent to the wrong person, possibly allowing unintended parties
+to generate signatures.
```
Comment (dconnolly) [inline book/src/tutorial/trusted-dealer.md:58]: Very nice.

File: book/src/tutorial/trusted-dealer.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+by simply using the secret to sign** (and this can't be
+possibly avoided with this method; use Distributed Key Generation
+if that's an issue), the dealer could also tamper with the `SecretShare`s
+in a way that the participants will never be able to generate a valid
+signature in the future (denial of service). Participants can detect
+such tampering by comparing the `VerifiableSecretSharingCommitment`
+values from their `SecretShare`s (either by some manual process, or
+by using a [broadcast channel](https://frost.zfnd.org/terminology.html#broadcast-channel))
+to make sure they are all equal.
+```
```
Comment (dconnolly) [inline book/src/tutorial/trusted-dealer.md:81]: 👍

**Review by dconnolly — DISMISSED** [review]
Looks good!

**Review by dconnolly — APPROVED** [review]
Looks good!

**Pull request merged**